nemo/collections/asr/losses/rnnt.py

```diff
@@ -281,19 +281,21 @@ def forward(self, log_probs, targets, input_lengths, target_lengths):
             targets = targets.narrow(dim=1, start=0, length=max_targets_len)
 
         # Loss reduction can be dynamic, so set it prior to call
-        if self.reduction != 'mean_batch':
+        if self.reduction in ['mean', 'sum']:  # underlying losses can do mean or sum
             self._loss.reduction = self.reduction
 
         # Compute RNNT loss
         loss = self._loss(acts=log_probs, labels=targets, act_lens=input_lengths, label_lens=target_lengths)
 
         # Loss reduction can be dynamic, so reset it after call
-        if self.reduction != 'mean_batch':
+        if self.reduction in ['mean', 'sum']:
             self._loss.reduction = 'none'
 
-        # Loss reduction only for mean_batch mode
+        # Loss reduction only for mean_batch and mean_volume modes
         if self.reduction == 'mean_batch':
             loss = torch.mean(loss)
```

Comment:
What is the difference between "mean" and "mean_batch" for this loss? In the underlying loss mean reduction seems to already be described as "the output losses will be divided by the target lengths and then the mean over the batch is taken"

Reply:
"mean_batch" - our current default - does not divide the losses by target lengths...